components/channel_layout/channel_controller.jsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Route} from 'react-router-dom';
+
+import Pluggable from 'plugins/pluggable';
+import AnnouncementBar from 'components/announcement_bar';
+import DeletePostModal from 'components/delete_post_modal.jsx';
+import EditPostModal from 'components/edit_post_modal';
+import GetPostLinkModal from 'components/get_post_link_modal';
+import GetTeamInviteLinkModal from 'components/get_team_invite_link_modal';
+import GetPublicLinkModal from 'components/get_public_link_modal';
+import InviteMemberModal from 'components/invite_member_modal.jsx';
+import LeaveTeamModal from 'components/leave_team_modal.jsx';
+import LeavePrivateChannelModal from 'components/modals/leave_private_channel_modal.jsx';
+import RemovedFromChannelModal from 'components/removed_from_channel_modal.jsx';
+import ResetStatusModal from 'components/reset_status_modal';
+import ShortcutsModal from 'components/shortcuts_modal.jsx';
+import SidebarRight from 'components/sidebar_right';
+import SidebarRightMenu from 'components/sidebar_right_menu';
+import TeamSettingsModal from 'components/team_settings_modal.jsx';
+import ImportThemeModal from 'components/user_settings/import_theme_modal.jsx';
+import UserSettingsModal from 'components/user_settings/user_settings_modal.jsx';
+import WebrtcNotification from 'components/webrtc/components/webrtc_notification.jsx';
+import WebrtcSidebar from 'components/webrtc/components/webrtc_sidebar.jsx';
+import ModalController from 'components/modal_controller';
+import TeamSidebar from 'components/team_sidebar';
+import Sidebar from 'components/sidebar';
+import * as Utils from 'utils/utils';
+import CenterChannel from 'components/channel_layout/center_channel';
+
+export default class ChannelController extends React.Component {
+    static propTypes = {
+        pathName: PropTypes.string.isRequired,
+        teamType: PropTypes.string.isRequired
+    };
+    shouldComponentUpdate(nextProps) {
+        return this.props.teamType !== nextProps.teamType || this.props.pathName !== nextProps.pathName;
+    }
```

Comment:
This could probably become a pure component since they're just primitive types

Reply:
Definitely. Plus, that would save having to remember to update this statement in the future if we add more props

Reply:
Reason I didn't is because the one prop is there only for shouldComponentUpdate. So I felt like the shouldComponentUpdate explains why that prop exists. Without it I will have to ignore-eslint a bunch of style mistakes

Reply:
In that case, do we need the pathName at all? I think the child components should be smart enough to re-render when the team/channel changes

Reply:
No. There was a bug when I had this a a PureComponent that saw clicking the different channels not changing anything. The issue was that the component was not rerendering because the teamType didn't change. 
adding the pathName in tells the component to recheck every time the url changes, which catches those channel switches.